source/simulator/helper_functions.cc

```diff
@@ -909,8 +909,166 @@ namespace aspect
 
     return (material_model->get_model_dependence().viscosity != MaterialModel::NonlinearDependence::none);
   }
-}
 
+
+
+  template <int dim>
+  void Simulator<dim>::apply_limiter_to_dg_solutions (const AdvectionField &advection_field)
+  {
+    AssertThrow (dim == 2,
+                 ExcMessage ("Only apply bound preserving limiter to the 2 dimensional problem"));
```

Comment:
If you want to generalize it to 3d at one point, please talk to me before you start implementing. I think I can see ways to make most of the code generic enough so that it requires only a few changes.